### PR TITLE
Automatic layout provider

### PR DESCRIPTION
**Pull request by thSoft**
#167

### CODE REVIEW

**Comment by thSoft** [comment]
To see the output of the algorithm, run ElkLayoutProviderTest from Eclipse. Currently it cannot be run with Tycho.

File: plugins/hu.bme.mit.massif.simulink.layout.elk/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+Bundle-SymbolicName: hu.bme.mit.massif.simulink.layout.elk
+Bundle-Version: 0.7.0.qualifier
+Automatic-Module-Name: hu.bme.mit.massif.simulink.layout.elk
+Bundle-RequiredExecutionEnvironment: JavaSE-1.8
+Bundle-Vendor: Budapest University of Technology and Economics, Fault Tolerant Systems Research Group & Embraer S.A.
+Require-Bundle: hu.bme.mit.massif.simulink.api;bundle-version="0.7.0",
+ org.eclipse.elk.graph;bundle-version="0.4.1",
+ org.eclipse.elk.core;bundle-version="0.4.1",
+ org.eclipse.elk.alg.layered;bundle-version="0.4.1",
+ com.google.guava;bundle-version="21.0.0"
```
Comment (abelhegedus) [inline plugins/hu.bme.mit.massif.simulink.layout.elk/META-INF/MANIFEST.MF]: Please remove Guava usage.

File: plugins/hu.bme.mit.massif.simulink.layout.elk/src/hu/bme/mit/massif/simulink/layout/elk/ElkLayoutProvider.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+            return ImmutableSet.of();
+        }
+    }
+    
+    public static Block getTarget(SingleConnection singleConnection) {
+        return singleConnection.getTo().getContainer();
+    }
+
+    private static Iterable<ElkNode> findNodes(Block block, Iterable<ElkNode> nodes) {
+        return Iterables.filter(nodes, node -> getBlock(node) == block);
```
Comment (abelhegedus) [inline plugins/hu.bme.mit.massif.simulink.layout.elk/src/hu/bme/mit/massif/simulink/layout/elk/ElkLayoutProvider.java]: I know we don't expect thousands of blocks, but wouldn't it be better to create a `Map<Block,ElkNode>` instead?

File: setup/hu.bme.mit.massif.setup/Massif.setup [hunk 3]
```diff
@@ -500,6 +500,10 @@
             name="com.sun.jna"/>
         <requirement
             name="com.sun.jna.platform"/>
+        <requirement
+            name="org.eclipse.elk.feature.source.feature.group"/>
```
Comment (abelhegedus) [inline setup/hu.bme.mit.massif.setup/Massif.setup]: Why are these source features? Oomph grabs sources automatically.

File: tests/hu.bme.mit.massif.simulink.layout.elk.tests/results/.gitignore [hunk 4]
```diff
@@ -0,0 +1,6 @@
+/one block.png
```
Comment (abelhegedus) [inline tests/hu.bme.mit.massif.simulink.layout.elk.tests/results/.gitignore]: How about `*.png` or even
```
*
!.gitignore
```

**Pull request merged**